profiles/commands.py

```diff
@@ -87,5 +91,7 @@ def _convert_orcid_date(orcid_date: dict) -> Optional[Date]:
         year = int(orcid_date['year']['value'])
         month = int(orcid_date['month']['value']) if orcid_date.get('month') else None
         day = int(orcid_date['day']['value']) if orcid_date.get('day') else None
-
-        return Date(year, month, day)
+        try:
+            return Date(year, month, day)
+        except ValueError as exception:
+            LOGGER.error('%s: %s', exception, (year, month, day))
```

Comment:
thus changes the behavior as unless `exception` is re-raised it will return `None`?

Reply:
This is only used to set a keyword argument value: `starts=_convert_orcid_date(orcid_affiliation.get('start-date') or {}),`. As the caller expects a None value if the year is not set, I think raising an exception here shouldn't be breaking. What do you think?

Reply:
If the caller can handle `None`, this is an actual bug fix probably as I agree in case of bad data coming in the best we can do is to log errors and skip those values (I remember someone seeing a 31st November or similar date). "breaking" is subtle as this was possibly raising an exception before (`ValueError`) and now won't be, it's a change that is probably for the best in this case.

(in general we tend to separate PRs by topic to keep them independent, improve logging vs. bugfix)

Reply:
Agreed this should probably be in a separate PR. Would you like me to remove it and create a new PR?

Reply:
Fine to keep it here for now, look out for similar cases into the future